### PR TITLE
Updating SQL vnet rules and tests with migration flag and state

### PR DESCRIPTION
**Pull request by kisantia**
- Updating SQL vnet rules with most recent swagger (https://github.com/Azure/azure-rest-api-specs/pull/1713)
- Updated and rerecorded vnet tests

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by markcowl — CHANGES_REQUESTED** [review]
I think you want to update the product version, or is this one of several PRs?

**Comment by jaredmoo** [comment]
1.7.0 is still not published yet so necessity of version bump depends on the timing of when we can publish the existing code

**Comment by markcowl** [comment]
@jaredmoo It is published now, so please update the version accordingly

File: src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj [hunk 1]
```diff
@@ -4,10 +4,15 @@
...
     <AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-    <VersionPrefix>1.7.0-preview</VersionPrefix>
+    <VersionPrefix>1.8.0-preview</VersionPrefix>
     <PackageTags>Microsoft Azure SQL Management;SQL;SQL Management</PackageTags>
   </PropertyGroup>
   <PropertyGroup>
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
+    <PackageReleaseNotes>
+      <![CDATA[
+        Added support for state and migration flag properties for SQL Vnet rules
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj]: minor issue: remove whitespace at start of the line, e.g.

```
      <![CDATA[
New features:
- Added support for state and migration flag properties for SQL Vnet rules
```

**Review by jaredmoo — APPROVED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**